diff --git a/plugins/meson.build b/plugins/meson.build
index c8c10db..30c38ab 100644
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -20,6 +20,8 @@ all_plugins = [
 
 disabled_plugins = []
 
+disabled_plugins += ['sharing']
+
 if not enable_smartcard
     disabled_plugins += ['smartcard']
 endif
@@ -33,7 +35,7 @@ if not enable_wacom
 endif
 
 if not enable_cups
-    disabled_plugins += ['cups']
+    disabled_plugins += ['print-notifications']
 endif
 
 if not enable_rfkill
diff --git a/plugins/power/meson.build b/plugins/power/meson.build
index 69e619f..cd1c88b 100644
--- a/plugins/power/meson.build
+++ b/plugins/power/meson.build
@@ -102,49 +102,4 @@ if host_is_linux
   )
 endif
 
-output = 'gsdpowerconstants.py'
-
-gsdpowerconstants_py = custom_target(
-  output,
-  input: 'gsd-power-constants.h',
-  output: output,
-  build_by_default: true,
-  command: [join_paths(meson.current_source_dir(), 'gsd-power-constants-update.pl'), '@INPUT@', '@OUTPUT@']
-)
-
-output = 'gsdpowerenums.py'
-
-gsdpowerenums_py = custom_target(
-  output,
-  output: output,
-  capture: true,
-  build_by_default: true,
-  command: [gsd_power_enums_update]
-)
-
-test_py = find_program('test.py')
-
-envs = environment()
-#envs.prepend('G_DEBUG', 'fatal-warnings')
-envs.set('BUILDDIR', meson.current_build_dir())
-envs.set('TOP_BUILDDIR', meson.build_root())
-envs.set('LD_PRELOAD', 'libumockdev-preload.so.0')
-envs.set('NO_AT_BRIDGE', '1')
-envs.set('HAVE_SYSFS_BACKLIGHT', host_is_linux ? '1' : '0')
-
-if get_option('b_sanitize').split(',').contains('address')
-  # libasan needs to be loaded first; so we need to explicitly preload it
-  envs.set('POWER_LD_PRELOAD', 'libasan.so.5')
-endif
-
-foreach i : [ 1, 2, 3, 4, 5, 6, 7, 8 ]
-  test(
-    'test-power @0@/8'.format(i),
-    test_py,
-    args: [ 'PowerPluginTest@0@'.format(i) ],
-    env: envs,
-    # The first set of tests takes very long because of SCREENSAVER_TIMEOUT_BLANK
-    timeout: i == 1 ? 180 : 120
-  )
-endforeach
 
